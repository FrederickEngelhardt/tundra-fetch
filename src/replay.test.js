describe('replay functionality', () => {
  it('should do something', () => {

  });
});
