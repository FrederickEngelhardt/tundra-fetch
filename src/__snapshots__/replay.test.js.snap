// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replay default should catch unmatched requests with debugging enabled 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": undefined,
        "headers": undefined,
        "method": "POST",
        "url": "http://someapi/api/user",
      },
      "response": Object {
        "content": "some content",
        "headers": Object {
          "header-one": Array [
            "one",
          ],
          "header-two": Array [
            "two",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    false,
  ],
]
`;

exports[`replay default should catch unmatched requests without debugging enabled 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "some content",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
        "header-two": Array [
          "two",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
]
`;

exports[`replay default should mock requests for a profile with a single request 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "some content",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
        "header-two": Array [
          "two",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
]
`;

exports[`replay default should mock requests for a profile with two requests and a repeat mode of 'first' 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
]
`;

exports[`replay default should mock requests for a profile with two requests and a repeat mode of 'last' 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
]
`;

exports[`replay default should mock requests for a profile with two requests and the default repeat mode 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
]
`;

exports[`replay default should mock requests for an profile with a single request 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "some content",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
        "header-two": Array [
          "two",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
]
`;

exports[`replay default should mock requests for an profile with two requests and a repeat mode of 'first' 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
]
`;

exports[`replay default should mock requests for an profile with two requests and a repeat mode of 'last' 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
    },
  ],
]
`;

exports[`replay default should mock requests for an profile with two requests and the default repeat mode 1`] = `
Array [
  Array [
    [Function],
    Object {
      "body": "content one",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
  Array [
    [Function],
    Object {
      "body": "content two",
      "headers": Object {
        "header-one": Array [
          "one",
        ],
      },
      "status": 200,
    },
    Object {
      "name": -2450936388,
      "overwriteRoutes": false,
      "repeat": 1,
    },
  ],
]
`;

exports[`replay matchingFunction should match a standard request on all factors and a matching config with debugging enabled 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should match a standard request on all factors and a matching config with headers to omit 1`] = `Array []`;

exports[`replay matchingFunction should match a standard request on all factors and a null matching config 1`] = `Array []`;

exports[`replay matchingFunction should match a standard request on all factors and an empty matching config 1`] = `Array []`;

exports[`replay matchingFunction should match a standard request on all factors with a fuzzy body 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should match a standard request on all factors with a fuzzy header 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should match a standard request on all factors with a fuzzy url 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should not match a standard request because the URL doesn't match 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should not match a standard request because the body doesn't match 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should not match a standard request because the headers don't match 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;

exports[`replay matchingFunction should not match a standard request because the method doesn't match 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": "body",
        "headers": Object {
          "abc": Array [
            "123",
          ],
          "xyz": Array [
            "456",
          ],
        },
        "method": "GET",
        "url": "http://www.someurl.com",
      },
      "response": Object {
        "content": "body",
        "headers": Object {
          "1": Array [
            "1",
          ],
          "2": Array [
            "2",
          ],
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
  ],
]
`;
